osvr/RenderKit/RenderManagerBase.cpp

```diff
@@ -1433,188 +1463,407 @@ namespace renderkit {
-                predictionIntervalSec, m_roomFromHead);
-            }
-
-            // Bring the pose into quatlib world.
-            q_from_OSVR(q_roomFromHead, m_roomFromHead);
-        }
-        q_xyz_quat_type q_roomFromEye;
-        q_xyz_quat_compose(&q_roomFromEye, &q_roomFromHead, &q_headFromEye);
-
-        // See if we are making a transform for world space.
-        // If don't have a callback defined for this space, we're in
-        // world space.  This is used by GetRenderInfo() and
-        // PresentRenderBuffers() to get its world-space matrix.
-        // If we have a NULL interface pointer, we are
-        // in world space.
-        bool inWorldSpace = (whichSpace >= m_callbacks.size()) ||
-                            (m_callbacks[whichSpace].m_interface == nullptr);
-
-        /// Include the impact of roomFromWorld, if it is specified.
-        /// If we are not going into world space, but rather into one
-        /// of the other OSVR spaces, then just leave this as the identity
-        /// transform so we don't need to undo it again on the way
-        /// back from room space.
-        q_xyz_quat_type q_worldFromRoom;
-        makeIdentity(q_worldFromRoom);
-        if (inWorldSpace && (params.worldFromRoomAppend != nullptr)) {
-            q_from_OSVR(q_worldFromRoom, *params.worldFromRoomAppend);
-        }
-        q_xyz_quat_type q_worldFromEye;
-        q_xyz_quat_compose(&q_worldFromEye, &q_worldFromRoom, &q_roomFromEye);
-
-        /// Invert the above matrices, to produce eyeFromWorld.
-        q_xyz_quat_type q_eyeFromWorld;
-        q_xyz_quat_invert(&q_eyeFromWorld, &q_worldFromEye);
-
-        /// Include the impact of the space we're rendering to.
-        /// This is spaceFromRoom; put on the right and multiply it on
-        /// the left by the above inverted matrix.  (If we are going
-        /// into one of these spaces, worldFromRoom will be the
-        /// identity so we don't need to invert and reapply it.)
-        q_xyz_quat_type q_worldFromSpace;
-        if (inWorldSpace) {
-            makeIdentity(q_worldFromSpace);
-        } else {
-            OSVR_TimeValue timestamp;
-            if (osvrGetPoseState(
-                    m_callbacks[whichSpace].m_interface, &timestamp,
-                    &m_callbacks[whichSpace].m_state) == OSVR_RETURN_FAILURE) {
-                // They asked for a space that does not exist.  Return false to
-                // let them know we didn't get the one they wanted.
-                return false;
-            }
-            q_from_OSVR(q_worldFromSpace, m_callbacks[whichSpace].m_state);
-        }
-        q_xyz_quat_type q_eyeFromSpace;
-        q_xyz_quat_compose(&q_eyeFromSpace, &q_eyeFromWorld, &q_worldFromSpace);
-
-        /// Store the result into the output pose
-        OSVR_from_q(eyeFromSpace, q_eyeFromSpace);
-        return true;
+		// If eye poses are present, use them instead of head +- IPD
```

Comment:
Looks like this got reflowed with tabs rather than spaces so the diff is not so helpful.